packages/animation-curve-block/src/components/AnimationCurveFlyout.tsx

```diff
@@ -137,75 +134,76 @@ export const AnimationCurveFlyout = ({
                         <div className="tw-relative tw-w-[74px]">
-                            <span className="tw-absolute tw-top-2.5 tw-left-[3px] tw-z-10 tw-text-s tw-text-text-weak">
-                                Y
-                            </span>
                             <TextInput
+                                onFocus={() => setZIndex(0)}
+                                onBlur={() => setZIndex(1000)}
                                 onChange={(value) => updateAnimationCurveParameters({ y1: Number(value) || 0 })}
                                 type={TextInputType.Number}
-                                value={localAnimationFunction.parameters.y1 as unknown as string}
+                                value={localAnimationFunction.parameters.y1.toString()}
                                 required
+                                decorator="Y"
                             />
                         </div>
                         <IconCaretRight16 />
                         <div className="tw-relative tw-w-[74px]">
-                            <span className="tw-absolute tw-top-2.5 tw-left-[3px] tw-z-10 tw-text-s tw-text-text-weak">
-                                X
-                            </span>
                             <TextInput
+                                onFocus={() => setZIndex(0)}
+                                onBlur={() => setZIndex(1000)}
                                 onChange={(value) => updateAnimationCurveParameters({ x2: Number(value) || 0 })}
                                 type={TextInputType.Number}
                                 value={localAnimationFunction.parameters.x2.toString()}
                                 validation={x2Validation}
                                 required
+                                decorator="X"
                             />
                         </div>
                         <div className="tw-relative tw-w-[74px]">
-                            <span className="tw-absolute tw-top-2.5 tw-left-[3px] tw-z-10 tw-text-s tw-text-text-weak">
-                                Y
-                            </span>
                             <TextInput
+                                onFocus={() => setZIndex(0)}
+                                onBlur={() => setZIndex(1000)}
                                 onChange={(value) => updateAnimationCurveParameters({ y2: Number(value) || 0 })}
                                 type={TextInputType.Number}
                                 value={localAnimationFunction.parameters.y2.toString()}
                                 required
+                                decorator="Y"
                             />
                         </div>
                     </div>
                     <div className="tw-w-20 tw-relative">
                         <TextInput
+                            onFocus={() => setZIndex(0)}
+                            onBlur={() => setZIndex(1000)}
```

Comment:
I'm sure I'm missing something but if the user starts to edit the input value, shouldn't it have higher zIndex than previously to be on top?

Reply:
Well the zIndex which get set is for the container around the animation curve and not for the TextInput, so onFocus the  the animation curve (basically just the handles can overflow) gets behind the TextInputs. But as its not clear I will adjust the naming a bit so its hopefully clearer